src/python_testing/TC_DGGEN_2_4.py

```diff
@@ -0,0 +1,192 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import asyncio
+import logging
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.interaction_model import InteractionModelError
+from matter_testing_support import (MatterBaseTest, async_test_body, default_matter_test_main, matter_epoch_us_from_utc_datetime,
+                                    utc_datetime_from_matter_epoch_us, utc_datetime_from_posix_time_ms)
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class TC_DGGEN_2_4(MatterBaseTest):
+    async def read_diags_attribute_expect_success(self, attribute):
+        cluster = Clusters.Objects.GeneralDiagnostics
+        return await self.read_single_attribute_check_success(endpoint=0, cluster=cluster, attribute=attribute)
+
+    async def read_timesync_attribute_expect_success(self, attribute):
+        cluster = Clusters.Objects.TimeSynchronization
+        return await self.read_single_attribute_check_success(endpoint=0, cluster=cluster, attribute=attribute)
+
+    async def set_time_in_timesync(self, current_time: int):
+        endpoint = 0
+        time_cluster = Clusters.Objects.TimeSynchronization
+        code = 0
+
+        try:
+            await self.send_single_cmd(cmd=time_cluster.Commands.SetUTCTime(UTCTime=current_time, granularity=time_cluster.Enums.GranularityEnum.kSecondsGranularity), endpoint=endpoint)
```

Comment:
```suggestion
            await self.send_single_cmd(cmd=time_cluster.Commands.SetUTCTime(UTCTime=current_time, granularity=time_cluster.Enums.GranularityEnum.kMillisecondsGranularity), endpoint=endpoint)
```
^ explanation: that's the controller granularity, if you send in seconds, the device might not trust your time.

Reply:
That's going to need a test plan change too, though.

Reply:
Will follow-up once testplan is fixed.